src/core/dep-graph.ts

```diff
@@ -183,6 +183,16 @@ class DepGraphImpl implements types.DepGraphInternal {
     return this.nodeEquals(this, this.rootNodeId, otherDepGraph, otherDepGraph.rootNodeId, compareRoot);
   }
 
+  public directDepsLeadingTo(pkg: types.Pkg): types.PkgInfo[] {
+    const pkgNodes = this.getPkgNodeIds(pkg);
+    const directDeps = this.getNodeDepsNodeIds(this.rootNodeId);
+    const nodes = directDeps.filter((directDep) => {
+      const reachableNodes = graphlib.alg.postorder(this._graph, [directDep]);
```

Comment:
please check (and add a test) if `graphlib.alg.postorder` works with cyclic graphs. If not, please check for cycles and throw if exist like done in some other places in the lib.
This also means this method will need to be re-implemented differently as part of https://github.com/snyk/dep-graph/pull/48 (but it's OK)

Reply:
It seems to work :D